src/main/java/com/mg105/use_cases/OpponentSetInteractor.java

```diff
@@ -54,6 +62,7 @@ public boolean setOpponentSet() {
         OpponentSet opponent = verifyOpponent();
 
         state.setCurrOpponent(opponent);
+        state.getCurrentRoom().getOpponents().remove(opponent);
```

Comment:
same issue here as with ChestInteractor, would this be problematic if opponent is null?

Reply:
It should be fine.